astropy/coordinates/baseframe.py

```diff
@@ -1366,6 +1366,58 @@ def sphericalcoslat(self):
         return self.represent_as('spherical', 'sphericalcoslat',
                                  in_frame_units=True)
 
+    @property
+    def velocity(self):
+        """
+        Shorthand for the Cartesian velocity as a `CartesianDifferential`
+        object.
+        """
+        if 's' not in self.data.differentials:
+            raise ValueError('Frame has no associated velocity (Differential) '
+                             'data information.')
+
+        try:
+            v = self.cartesian.differentials['s']
+        except Exception as e:
+            raise ValueError('Could not retrieve a Cartesian velocity. Your '
```

Comment:
I think this error is a bit confusing. I can have a coordinate with only PMs and no radial velocity *and* no distance and this will still work fine (and it *should*).  Maybe instead "Your frame data must include velocity information for this to work"?